### DISCUSSION TITLE
Make `require` work in ESM bundle

### DISCUSSION THREAD

**Discussion by egoist** [egoist/tsup#505] [Ideas]
If you're building in esm format, and one of your dependency is written in CJS and you want to bundle it, the result code might contain `require` call:

```js
var fs = require('fs')
```

That's because built-in modules can't be resolved and esbuild just leaves it as is, a workaround is to use the `banner.js` option to make `require` available in your esm bundle:

```ts
// tsup.config.ts
import { defineConfig } from "tsup";

export default defineConfig({
  banner: {
    js: `import {createRequire as __createRequire} from 'module';var require=createRequire(import\.meta.url);`,
  },
});
```

**Comment by ModyQyW** [comment]
```typescript
// tsup.config.ts
import { defineConfig } from "tsup";

export default defineConfig({
  banner: {
    js: `import {createRequire as __createRequire} from 'module';var require=__createRequire(import\.meta.url);`,
  },
});
```

Shouldn't we use `__createRequire` in `var require=createRequire(import\.meta.url);`?
